common-rendering/src/components/keyEvents.tsx

```diff
@@ -0,0 +1,187 @@
+// ----- Imports ----- //
+
+import { css } from "@emotion/react";
+import type { SerializedStyles } from "@emotion/react";
+import { FC } from "react";
+import { textSans, headline } from "@guardian/src-foundations/typography";
+import { palette, remSpace } from "@guardian/src-foundations";
+import { SvgChevronUpSingle, SvgChevronDownSingle } from "@guardian/src-icons";
+import { Pillar, Theme } from "@guardian/types";
+import { from } from "@guardian/src-foundations/mq";
+
+// ----- Component ----- //
+type paletteId = 300 | 400;
+
+interface KeyEvent {
+	time: string;
+	text: string;
+	url: string;
+}
+
+interface KeyEventsProps {
+	keyEvents: KeyEvent[];
+	theme: Theme;
+}
+
+interface ListItemProps {
+	keyEvent: KeyEvent;
+	theme: Theme;
+}
+
+const getColor = (theme: Theme, paletteId: paletteId) => {
+	switch (theme) {
+		case Pillar.Sport:
+			return palette.sport[paletteId];
+		case Pillar.Culture:
+			return palette.culture[paletteId];
+		case Pillar.Lifestyle:
+			return palette.lifestyle[paletteId];
+		case Pillar.Opinion:
+			return palette.opinion[paletteId];
+		default:
+			return palette.news[paletteId];
+	}
+};
+
+const keyEventWrapperStyles: SerializedStyles = css`
+	width: 100%;
+
+	${from.desktop} {
+		width: 13.75rem;
+	}
+`;
+
+const detailsStyles: SerializedStyles = css`
+	&:not([open]) .is-on,
+	&[open] .is-off {
+		display: none;
+	}
+
+	summary::-webkit-details-marker {
+		display: none;
+	}
+`;
+
+const titleRowStyles: SerializedStyles = css`
+	position: relative;
+	margin: ${remSpace[1]} ${remSpace[1]} 0 ${remSpace[3]};
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+
+	&:focus {
+		outline: none;
```

Comment:
Would this cause problems for people using keyboard navigation?

Reply:
+1 I think even if it looks clunky we should preserve focus highlighting for accessibility (and also just for semantics)

Reply:
I know you mentioned in DCR the `focusHalo` is added as a style in Global. But since we don't yet have this in apps-rendering, I used it within the keyevent component for the focus effect:
```
&:focus {
		${focusHalo}
	}
```

What do you think?

Reply:
Looks great 👍